server/resp/src/main/java/org/infinispan/server/resp/ByteBufferUtils.java

```diff
@@ -46,6 +46,20 @@ private ByteBufferUtils() {
          '0', '1', '2', '3', '4', '5', '6', '7', '8', '9',
    };
 
+   public static ByteBuf stringToResult(byte[] result, ByteBufPool alloc) {
+      int length = result.length;
+      int stringLength = stringSize(length);
+      int exactSize = stringLength + 1;
+      ByteBuf buffer = alloc.acquire(exactSize);
+      buffer.writeByte('+');
+      if (result!=null) {
```

Comment:
`result` can be null? If so, we need to add this at the first line of the method to avoid the NPE on line 50 with `result.length` with something like `if (result == null) return alloc.acquire(NIL.length).writeBytes(NIL);`

Reply:
seems that RESP allows empty string but not nil. only bulk string has a specific encoding for nil. https://amitshekhar.me/blog/resp-redis-serialization-protocol

Reply:
I see. So we can update this to:

```java
int exactSize = result == null ? 1 + CRLF.length : 1 + stringSize(result.length) + CRLF.length;
```

And we can leave the remaining as is.